### ISSUE TITLE
Upgrade from 2.10.1 to 2.10.2 causes "Class constructor CoreMongooseArray cannot be invoked without 'new'"

### ISSUE THREAD

**Issue by amit777** [node-cache-manager/node-cache-manager#134]
Below is the stack trace. i can reproducably fix the problem if i downgrade to 2.10.1.  If I go to 2.10.2 or later, the below error happens.

I'm creating a mongoose object using memory cache.  Here is a code snippet

```
var myCache = cacheManager.caching({store: 'memory', max: _memoryMax, ttl: _memoryTTL});
var Domain = mongoose.model('Domain');

var d = await myCache.wrap('mykey', async function() {
   var ret = new Domain(pojo);
  return ret;
})
```

Stack trace
```
2020-02-05T17:59:39.834Z[32290] - debug: TypeError: Class constructor CoreMongooseArray cannot be invoked without 'new'
    at initCloneArray (/opt/app/node_modules/lodash.clonedeep/index.js:1235:22)
    at baseClone (/opt/app/node_modules/lodash.clonedeep/index.js:854:14)
    at /opt/app/node_modules/lodash.clonedeep/index.js:897:30
    at arrayEach (/opt/app/node_modules/lodash.clonedeep/index.js:140:9)
    at baseClone (/opt/app/node_modules/lodash.clonedeep/index.js:891:3)
    at /opt/app/node_modules/lodash.clonedeep/index.js:897:30
    at arrayEach (/opt/app/node_modules/lodash.clonedeep/index.js:140:9)
    at baseClone (/opt/app/node_modules/lodash.clonedeep/index.js:891:3)
    at /opt/app/node_modules/lodash.clonedeep/index.js:897:30
    at arrayEach (/opt/app/node_modules/lodash.clonedeep/index.js:140:9)
    at baseClone (/opt/app/node_modules/lodash.clonedeep/index.js:891:3)
    at cloneDeep (/opt/app/node_modules/lodash.clonedeep/index.js:1399:10)
    at Object.self.set (/opt/app/node_modules/cache-manager/lib/stores/memory.js:35:17)
    at /opt/app/node_modules/cache-manager/lib/multi_caching.js:258:29
    at /opt/app/node_modules/cache-manager/node_modules/async/lib/async.js:181:20
    at Object.async.forEachOf.async.eachOf (/opt/app/node_modules/cache-manager/node_modules/async/lib/async.js:233:13)
    at Object.async.forEach.async.each (/opt/app/node_modules/cache-manager/node_modules/async/lib/async.js:209:22)
    at setInMultipleCaches (/opt/app/node_modules/cache-manager/lib/multi_caching.js:220:15)
    at /opt/app/node_modules/cache-manager/lib/multi_caching.js:396:41
    at /opt/app/node_modules/cache-manager/lib/multi_caching.js:314:21
    at processTicksAndRejections (internal/process/task_queues.js:93:5) - {}
```

**Comment by BryanDonovan** [comment]
OK, looks like we need to add an option to the memory store config.  However, I do believe that it's not the right use of a cache to be caching anything other than plain objects (basically anything that can be serialized as JSON or similar).

For now, please just use version 2.10.1.  I'll get this fixed as soon as I can, hopefully in the next couple of days.

**Comment by amit777** [comment]
Thanks Bryan!  You are probably right about not caching anything other than plain objects, but we are getting a tremendous boost here because constructing a mongoose object on every request to our web server causes a huge load.  Caching the constructed mongoose object was our knee-jerk fix a while ago, and we've kept it since it works pretty well.

I will ping to 2.10.1 for now.

**Comment by BryanDonovan** [comment]
Yeah, that makes sense.  The memory store has that interesting ability to cache any javascript object.

**Comment by RamyTemim** [comment]
Hi ! 
I've the same error. Downgrade from 2.11.1 to 2.10.1 fix the problem
Thank you

**Comment by BryanDonovan** [comment]
This is fixed in #135 (version 3.0.0) by allowing you to pass in `shouldCloneBeforeSet` in the `caching` options.  E.g.,

```javascript
const cache = caching({store: 'memory', shouldCloneBeforeSet: false});
```

**Comment by amit777** [comment]
Wow, sorry for the very late reply. I will upgrade and try this out! Thanks!
